Framework/Algorithms/src/ExtractMask.cpp

```diff
@@ -62,26 +63,21 @@ void ExtractMask::exec() {
 
   // Create a new workspace for the results, copy from the input to ensure
   // that we copy over the instrument and current masking
-  auto maskWS = std::make_shared<DataObjects::MaskWorkspace>(inputWS);
-  maskWS->setTitle(inputWS->getTitle());
+  std::shared_ptr<DataObjects::MaskWorkspace> maskWS;
+  if (getProperty("UngroupDetectors"))
+    maskWS = std::make_shared<DataObjects::MaskWorkspace>(inputWS->getInstrument());
+  else
+    maskWS = std::make_shared<DataObjects::MaskWorkspace>(inputWS);
 
-  const auto &spectrumInfo = inputWS->spectrumInfo();
-  const auto nHist = static_cast<int64_t>(inputWS->getNumberHistograms());
-  Progress prog(this, 0.0, 1.0, nHist);
+  maskWS->setTitle(inputWS->getTitle());
 
-  PARALLEL_FOR_IF(Kernel::threadSafe(*inputWS, *maskWS))
-  for (int64_t i = 0; i < nHist; ++i) {
-    PARALLEL_START_INTERRUPT_REGION
-    bool inputIsMasked(false);
-    if (spectrumInfo.hasDetectors(i)) {
-      // special workspaces can mysteriously have the mask bit set
-      inputIsMasked = (inputWSIsSpecial && inputMaskWS->isMaskedIndex(i)) || spectrumInfo.isMasked(i);
+  // set mask from from detectorList
```

Comment:
It looks like we went from an OMP-threaded for loop to a non-threaded for loop.  Was there no other way to do this?  (If I assume masking is _rare_, this is probably OK.  I can imagine the opposite case, however.)